### PR TITLE
feat(thumbnail): offset option

### PR DESCRIPTION
**Pull request by cwklausing**
[Card](https://rentpath.atlassian.net/browse/AL-35)

Allows us to customize the `perIndexScroll` for gallery thumbnails.

This PR is a dependency for [this](https://github.com/rentpath/ag.js/pull/5648) PR in AG.js

### CODE REVIEW

**Comment by tadjohnston** [comment]
I guess I'm a little confused on what you're trying to do here. Can you explain a little more? The linked ag.js card doesn't have any information on what this is for.

**Comment by cwklausing** [comment]
@tadjohnston -- We're updating the PDP gallery to use a thumbnail scroller. `react-image-gallery` already has a built-in thumbnail scroller, however, it doesn't scroll by exactly one slide. Here's a picture of how the current scrolling offset looks:

screen shot 2018-11-05 at 12 50 13 pm

and this is how it looks with this update (when the `lastThumbnailOffset` is set to 3):

screen shot 2018-11-05 at 12 52 08 pm

**Comment by tadjohnston** [comment]
Are you sure this isn't just a css issue?

**Comment by cwklausing** [comment]
Yes. I can use css to update the width of the photos, but the `state.thumbnailTranslate` is still calculated as a proportion of the entire containers width (which is simply #photos * width of each photo). So we need to actually change how the thumbnailTranslate is calculated.

**Comment by barelyfitz** [comment]
Is forking `react-image-gallery` the only way we can accomplish this? Because that would make it more difficult to upgrade if there are changes in the future. For example, if this is a useful feature, we could make a pull request against `react-image-gallery` and see if they would make it a standard feature.

**Comment by cwklausing** [comment]
@barelyfitz we're already using a forked version, so that convo is outside the scope of this PR

**Review by barelyfitz — APPROVED** [review]
👍 Seems okay to me.

**Comment by tadjohnston** [comment]
> the state.thumbnailTranslate is still calculated as a proportion of the entire containers width (which is simply #photos * width of each photo)

This sounds vaguely familiar to the same issue we've seen before with the photos themselves doing the same.  Adding `width: 100%` to the parents solved this issue. 

I would rather not fix the issue like this.  This doesn't address the problem (if there is one), with the gallery itself.  It is merely allowing you to bypass the issue by passing in a prop IMO.

**Comment by cwklausing** [comment]
@tadjohnston - don't want to merge this without your 👍 -- do you have any more concerns/questions?

**Comment by cwklausing** [comment]
I agree--I wish there was a solution that didn't alter the library, but I disagree that "this doesn't address the issue with the gallery": the issue as I see it is that the gallery only provides the option to scroll by one set proportion--and conversely has no way to scroll by **exactly** the width of one slide. I would love to see a CSS solution that doesn't require altering this library, however, I haven't found anything after a few days working with it.

File: build/image-gallery.js
Comment (tadjohnston) [inline build/image-gallery.js]: What happens if `totalThumbnails` is 2 and `lastThumbnailOffset` is 3?

File: src/ImageGallery.jsx
Comment (tadjohnston) [inline src/ImageGallery.jsx:642]: What happens if `totalThumbnails` is 2 and`lastThumbnailOffset` is 3? The `translate` will be a negative number.

**Review by tadjohnston — CHANGES_REQUESTED** [review]
I cannot approve this until we get this right. This repo is used all over ag.js _and_ rent _and_ rentals, so we need to make sure it's solid before we merge.

File: src/ImageGallery.jsx
Comment (mpelzsherman) [reply src/ImageGallery.jsx:642]: How about 

```javascript
 let perIndexScroll = Math.max(0, totalScroll / (totalThumbnails - lastThumbnailOffset));
```

**Pull request closed without merging**
